btcd.go

```diff
@@ -88,6 +88,17 @@ func btcdMain(serverChan chan<- *server) error {
 		defer pprof.StopCPUProfile()
 	}
 
+	// Write mem profile if requested.
+	if cfg.MemoryProfile != "" {
+		f, err := os.Create(cfg.MemoryProfile)
+		if err != nil {
+			btcdLog.Errorf("Unable to create memory profile: %v", err)
+			return err
+		}
+		defer f.Close()
+		defer pprof.WriteHeapProfile(f)
```

Comment:
since the file is only ever written to on shutdown, can't this whole if statement be put in a `defer`? that way we also dont need to keep the file open for the lifetime of the binary

Reply:
I think it'd be nice if it errors out immediately if it's not able to create a file to write to.

I'm not sure on the costs of having a file descriptor for the duration of the program but if that's a concern, how about something like this:

```
        // Write mem profile if requested.
        if cfg.MemoryProfile != "" {
                f, err := os.Create(cfg.MemoryProfile)
                if err != nil {
                        btcdLog.Errorf("Unable to create memory profile: %v", err)
                        return err
                }
                f.Close()

                defer func() {
                        f, err := os.Create(cfg.MemoryProfile)
                        if err != nil {
                                btcdLog.Infof("Unable to create memory profile: %v", err)
                        }
                        err = pprof.WriteHeapProfile(f)
                        if err != nil {
                                btcdLog.Infof("Unable to create memory profile: %v", err)
                        }
                        f.Close()
                }()
        }
```

Reply:
good point regarding failing fast on start up 👍  in that case, if the cost of the file descriptor is low then your original solution seems good enough!

Reply:
I think the cost is pretty low

Reply:
Ok I'll keep the current implementation.